pylint/checkers/exceptions.py

```diff
@@ -201,13 +200,29 @@ def visit_name(self, node: nodes.Name) -> None:
             self._checker.add_message(
                 "notimplemented-raised", node=self._node, confidence=HIGH
             )
-        elif node.name in OVERGENERAL_EXCEPTIONS:
-            self._checker.add_message(
-                "broad-exception-raised",
-                args=node.name,
-                node=self._node,
-                confidence=HIGH,
-            )
+            return
+
+        try:
+            exceptions = list(_annotated_unpack_infer(node))
+        except astroid.InferenceError:
+            return
+
+        for _, exception in exceptions:
+            if isinstance(exception, astroid.Instance) and utils.inherit_from_std_ex(
+                exception
+            ):
+                exception = exception._proxied
+
+            if not isinstance(exception, nodes.ClassDef):
+                continue
+
+            if self._checker._is_overgeneral_exception(exception):
+                self._checker.add_message(
+                    "broad-exception-raised",
+                    args=exception.name,
+                    node=self._node,
+                    confidence=HIGH,
```

Comment:
```suggestion
                    confidence=INFERENCE,
```

One final nit.. This would require updating the test output as well.